### PR TITLE
Adds basic github action for npm publish

### PR DESCRIPTION
**Pull request by rvwatch**
Simple github action based on the default NPM publish template that github provides.

### CODE REVIEW

File: .github/workflows/npmpublish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: NPM Publish
+
+on:
+  release:
+    types: [published]
```
Comment (rvwatch) [inline .github/workflows/npmpublish.yml:8]: @amphro There were several options around this but published seemed to make sense for at least this first iteration.

File: .github/workflows/npmpublish.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# This workflow will publish a package to NPM when a release is published
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: NPM Publish
```
Comment (lcampos) [inline .github/workflows/npmpublish.yml:4]: GitHub Actions are not approved by security yet, we're still in the review phase. This shouldn't be merged until we get their approval.

Comment (rvwatch) [reply .github/workflows/npmpublish.yml:4]: Ah, I wasn't aware of that. Ok.

**Comment by rvwatch** [comment]
Github actions are not approved for use in publishing packages.

**Pull request closed without merging**